benchmarks/torchvision_classification/train.py

```diff
@@ -0,0 +1,346 @@
+    if args.no_transforms:
+        train_preset = val_preset = helpers.no_transforms
+    else:
+        train_preset = presets.ClassificationPresetTrain(crop_size=train_crop_size)
+        val_preset = presets.ClassificationPresetEval(crop_size=val_crop_size, resize_size=val_resize_size)
+
+    if args.ds_type == "dp":
+        builder = helpers.make_pre_loaded_dp if args.preload_ds else helpers.make_dp
+        train_dataset = builder(train_dir, transforms=train_preset)
+        val_dataset = builder(val_dir, transforms=val_preset)
+
+        train_sampler = val_sampler = None
+        train_shuffle = True
+
+    elif args.ds_type == "iterable":
+        train_dataset = torchvision.datasets.ImageFolder(train_dir, transform=train_preset)
+        train_dataset = helpers.MapStyleToIterable(train_dataset, shuffle=True)
+
+        val_dataset = torchvision.datasets.ImageFolder(val_dir, transform=val_preset)
+        val_dataset = helpers.MapStyleToIterable(val_dataset, shuffle=False)
+
+        train_sampler = val_sampler = None
+        train_shuffle = None  # but actually True
+
+    elif args.ds_type == "mapstyle":
+        builder = helpers.PreLoadedMapStyle if args.preload_ds else torchvision.datasets.ImageFolder
+        train_dataset = builder(train_dir, transform=train_preset)
+        val_dataset = builder(val_dir, transform=val_preset)
+
+        train_sampler = torch.utils.data.distributed.DistributedSampler(train_dataset, shuffle=True)
+        val_sampler = torch.utils.data.distributed.DistributedSampler(val_dataset, shuffle=False)
+        train_shuffle = None  # but actually True
+
+    else:
+        raise ValueError(f"Invalid value for args.ds_type ({args.ds_type})")
+
+    data_loader_arg = args.data_loader.lower()
+    if data_loader_arg == "v1":
+        train_data_loader = torch.utils.data.DataLoader(
+            train_dataset,
+            batch_size=args.batch_size,
+            shuffle=train_shuffle,
+            sampler=train_sampler,
+            num_workers=args.workers,
+            pin_memory=True,
+            drop_last=True,
+        )
+        val_data_loader = torch.utils.data.DataLoader(
+            val_dataset,
+            batch_size=args.batch_size,
+            sampler=val_sampler,
+            num_workers=args.workers,
+            pin_memory=True,
+        )
+    elif data_loader_arg == "v2":
+        if args.ds_type != "dp":
+            raise ValueError("DataLoader2 only works with datapipes.")
+
+        # Note: we are batching and collating here *after the transforms*, which is consistent with DLV1.
+        # But maybe it would be more efficient to do that before, so that the transforms can work on batches??
+
+        train_dataset = train_dataset.batch(args.batch_size, drop_last=True).collate()
+        train_data_loader = DataLoader2(
+            train_dataset,
+            datapipe_adapter_fn=adapter.Shuffle(),
+            reading_service=MultiProcessingReadingService(num_workers=args.workers),
```

Comment:
Note that `MultiProcessingReadingService` will be updated to use a slightly different backend (i.e. what is currently `PrototypeMultiProcessingReadingService`.

I'm going to look into that part to see if there is any issue.